overlord/snapstate/handlers.go

```diff
@@ -1690,6 +1703,8 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 	}
 	if snapsup.MigratedToExposedHome {
```

Comment:
shouldn't this consider also EnableExposedHome ?

Reply:
should `if snapsup.MigratedToExposedHome`  in  snapsup.MigratedToExposedHome consider Disable/EnaleExposedHome ?


maybe we need a predicate function UseExposedHome  on SnapSetup?

Reply:
Yes, this doesn't have the enable/disable yet (well now it does, with the migration state helper you suggested) but ultimately it must also consider those flags